internal/demo/README.md

```diff
@@ -29,45 +29,73 @@ This guide will help you set up and run the Temporal Worker Controller locally u
      ```bash
      make create-cloud-mtls-secret
      ```
-   - Create a `skaffold.env` file:
+   - In `skaffold.example.env`, set:
+     ```env
+     TEMPORAL_API_KEY_SECRET_NAME=""
+     TEMPORAL_MTLS_SECRET_NAME=temporal-cloud-mtls-secret
+     ```
+
+   NOTE: Alternatively, if you are using API keys, follow the steps below instead of mTLS:
+
+   #### Using API Keys (alternative to mTLS)
+   - Create a `certs` directory in the project root if not already present
+   - Save your Temporal Cloud API key in a file (single line, no newline):
+     ```bash
+     echo -n "<YOUR_API_KEY>" > certs/api-key.txt
+     ```
+   - Create the Kubernetes Secret (trims any accidental newlines):
+     ```bash
+     make create-api-key-secret
+     ```
+   - In `skaffold.example.env`, set:
+     ```env
+     TEMPORAL_API_KEY_SECRET_NAME=temporal-api-key
+     TEMPORAL_MTLS_SECRET_NAME=""
+     ```
+   - Note: Do not set both mTLS and API key for the same connection. If both present, the controller 
```

Comment:
```suggestion
   - Note: Do not set both mTLS and API key for the same connection. If both are present, the controller 
```

Reply:
The behavior of the controller when two secrets are present is a good thing to think about. IMO, we should add a TemporalConnection validator that makes it so that TemporalConnections with both the API SecretRef and mTLS SecretRef field populated are rejected as invalid. Then the controller won't have to make the call about which to use. WDYT?